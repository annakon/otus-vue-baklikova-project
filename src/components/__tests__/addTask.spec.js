import {describe, it, vi, expect, beforeEach} from "vitest";
import {mount} from "@vue/test-utils";
import component from "@/components/addTask.vue";
import { setActivePinia, createPinia } from 'pinia'

vi.mock('vue-router')
describe('Home', () => {
    let wrapper
    const newdate = new Date("01.08.2023")

    beforeEach(() => {
        setActivePinia(createPinia())
        wrapper = mount(component, { global: {
                stubs: ['RouterLink'],
            },props: { data:newdate } });
    });

    it('renders properly', () => {
        expect(wrapper).toBeDefined();
        expect(wrapper.exists()).toBe(true);
    });
});
