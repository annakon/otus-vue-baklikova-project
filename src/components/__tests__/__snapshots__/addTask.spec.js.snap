// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > renders all 1`] = `
"<form data-v-943c0c07=\\"\\" data=\\"Sun Jan 08 2023 00:00:00 GMT+0300 (Москва, стандартное время)\\">
  <div data-v-943c0c07=\\"\\" class=\\"mb-3\\"><label data-v-943c0c07=\\"\\" for=\\"date\\" class=\\"form-label\\">Дата</label><input data-v-943c0c07=\\"\\" type=\\"date\\" class=\\"form-control\\" id=\\"date\\"></div>
  <div data-v-943c0c07=\\"\\" class=\\"mb-3\\"><label data-v-943c0c07=\\"\\" class=\\"form-label\\" for=\\"appt\\">Время</label><input data-v-943c0c07=\\"\\" class=\\"form-control\\" type=\\"time\\" id=\\"appt\\"></div>
  <div data-v-943c0c07=\\"\\" class=\\"mb-3\\"><label data-v-943c0c07=\\"\\" for=\\"description\\" class=\\"form-label\\">Описание</label><textarea data-v-943c0c07=\\"\\" class=\\"form-control\\" id=\\"description\\" rows=\\"3\\"></textarea></div>
  <div data-v-943c0c07=\\"\\" class=\\"mb-3\\"><button data-v-943c0c07=\\"\\" class=\\"btn btn-primary\\" type=\\"button\\">Добавить</button></div>
</form>"
`;

exports[`addTask > renders all 1`] = `
"<form data-v-943c0c07=\\"\\" data=\\"Sun Jan 08 2023 00:00:00 GMT+0300 (Москва, стандартное время)\\">
  <div data-v-943c0c07=\\"\\" class=\\"mb-3\\"><label data-v-943c0c07=\\"\\" for=\\"date\\" class=\\"form-label\\">Дата</label><input data-v-943c0c07=\\"\\" type=\\"date\\" class=\\"form-control\\" id=\\"date\\"></div>
  <div data-v-943c0c07=\\"\\" class=\\"mb-3\\"><label data-v-943c0c07=\\"\\" class=\\"form-label\\" for=\\"appt\\">Время</label><input data-v-943c0c07=\\"\\" class=\\"form-control\\" type=\\"time\\" id=\\"appt\\"></div>
  <div data-v-943c0c07=\\"\\" class=\\"mb-3\\"><label data-v-943c0c07=\\"\\" for=\\"description\\" class=\\"form-label\\">Описание</label><textarea data-v-943c0c07=\\"\\" class=\\"form-control\\" id=\\"description\\" rows=\\"3\\"></textarea></div>
  <div data-v-943c0c07=\\"\\" class=\\"mb-3\\"><button data-v-943c0c07=\\"\\" class=\\"btn btn-primary\\" type=\\"button\\">Добавить</button></div>
</form>"
`;
